warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IteratorBuildingVisitor.java

```diff
@@ -1047,21 +1047,22 @@ public Object visit(ASTNumberLiteral node, Object o) {
     private List<IvaratorCacheDir> getIvaratorCacheDirs() throws IOException {
         List<IvaratorCacheDir> pathAndFs = new ArrayList<>();
         
-        // first lets increment the count for a unique subdirectory
-        String subdirectory = ivaratorCacheSubDirPrefix + "term" + Integer.toString(++ivaratorCount);
+        StringBuilder subDirectoryBuilder = new StringBuilder();
+        subDirectoryBuilder.append(queryId);
+        if (scanId == null) {
+            log.warn("Running query iterator for " + queryId + " without a scan id.  This could cause ivarator directory conflicts.");
+        } else {
+            subDirectoryBuilder.append('_').append(scanId);
+        }
```

Comment:
This seems wrong to rely solely on an integer count for directory uniqueness. I don't have a sense for how frequent this is, but there might be a better way to guard against this case.

Maybe if the scanId is null, this could fall back to a date-based id? For example: yyyyMMddhhmmSS